charts/orchestrator/templates/sonataflows.yaml

```diff
@@ -1,22 +1,38 @@
-{{- $ns := .Values.orchestrator.namespace }}
-{{- if .Values.orchestrator.devmode }}
-  {{- $namespace := (lookup "v1" "Namespace" "" $ns) }}
-  {{- if $namespace }}
-# Namespace exist, do nothing.
-  {{- else }}
+{{- define "persistence" }}
+      persistence:
+        postgresql:
+          secretRef:
+            name: {{ .Values.postgres.authSecret.name }}
+            userKey: {{ .Values.postgres.authSecret.userKey }}
+            passwordKey: {{ .Values.postgres.authSecret.passwordKey }}
+          serviceRef:
+            name: {{ .Values.postgres.serviceName }}
+            namespace: {{ .Values.postgres.serviceNamespace }}
+{{- end }}
+
+{{- if .Values.sonataFlowOperator.enabled }}
+  {{- $ns := .Values.orchestrator.namespace }}
+  {{- if .Values.orchestrator.devmode }}
+---
 apiVersion: v1
 kind: Namespace
 metadata:
   name: {{ $ns }}
   labels:
+    {{- if .Values.argocd.enabled }}
     argocd.argoproj.io/managed-by: {{ .Values.argocd.namespace }}
----
+    {{- end }}
   {{- end }}
+---
 apiVersion: sonataflow.org/v1alpha08
 kind: SonataFlowPlatform
 metadata:
   name: sonataflow-platform
   namespace: {{ $ns }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-delete-policy": hook-fail
```

Comment:
Just a thought: should we add also `before-hook-creation` to avoid conflicts?

Reply:
I'm afraid that adding `before-hook-creation` will impact existing functionality as it will cause an outage while the installation/upgrade is being processed by removing the job and data index services. I think a new pod rollout is safer than removing the existing pods and deploying new ones. WDYT?

Reply:
There might be an issue with reattempting to install the workflow when an installation fails. But I'm okay with starting with the implementation and addressing issues if the pop-up.